main/main.cpp

```diff
@@ -691,12 +691,12 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 		if (I->get() == "-h" || I->get() == "--help" || I->get() == "/?") { // display help
 
 			show_help = true;
-			exit_code = OK;
+			exit_code = ERR_HELP; // Hack to force an early exit in `main()` with a success code.
```

Comment:
For the record we discussed this on chat, and chose to go with this hack using the pre-existing `ERR_HELP` (which nothing uses otherwise in the codebase) to convey the meaning of early exit but success.

It's not the prettiest but the whole `main` setup is pretty messy already and due a full refactor/rewrite, so in the meantime this is good enough because it's simple/non-intrusive.